### ISSUE TITLE
Crash when sending to gadgetbridge

### ISSUE THREAD

**Issue by kosmoz** [breezy-weather/breezy-weather#951]
### Steps to reproduce

1. Enable gadgedbridge in settings
2. Add location: Vienna
3. Use geosphere source
4. Refresh data

### Expected behavior

No crash

### Actual behavior

Crash

### Weather source used

Geosphere Austria

### Breezy Weather version

5.2.0

### Android version

13

### Device

Fairphone 4

### Other details

--------- beginning of main
04-16 06:59:21.573  6784  6784 E g.breezyweather: Not starting debugger since process cannot load the jdwp agent.
04-16 06:59:21.804  6784  6846 E SQLiteLog: (283) recovered 957 frames from WAL file /data/data/org.breezyweather/databases/breezyweather.db-wal
04-16 06:59:54.321  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:07.222  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:10.995  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:12.212  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:17.872  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:22.274  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:00:27.135  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:03.568  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:05.618  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:20.051  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:24.721  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:32.401  6784  6867 E OpenGLRenderer: Unable to match the desired swap behavior.
--------- beginning of crash
04-16 07:01:33.459  6784  7032 E AndroidRuntime: FATAL EXCEPTION: RxCachedThreadScheduler-1
04-16 07:01:33.459  6784  7032 E AndroidRuntime: Process: org.breezyweather, PID: 6784
04-16 07:01:33.459  6784  7032 E AndroidRuntime: P2.e: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at S2.b.a(SourceFile:10)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.observers.c.onError(Unknown Source:13)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onError(Unknown Source:37)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onComplete(Unknown Source:36)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.checkTerminated(SourceFile:61)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.drainNormal(Unknown Source:30)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.run(Unknown Source:8)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.q.run(Unknown Source:13)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.q.call(Unknown Source:0)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: Caused by: java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:595)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:480)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:909)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at org.breezyweather.common.extensions.e.G(SourceFile:7)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at y4.b.y(SourceFile:26)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at y4.b.A(SourceFile:918)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at org.breezyweather.sources.p.b(SourceFile:331)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at B0.o.run(SourceFile:141)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at h4.a.run(SourceFile:13)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onComplete(Unknown Source:7)
04-16 07:01:33.459  6784  7032 E AndroidRuntime: 	... 10 more
04-16 07:01:34.643  7394  7394 E g.breezyweather: Not starting debugger since process cannot load the jdwp agent.
04-16 07:01:34.767  7394  7421 E SQLiteLog: (283) recovered 91 frames from WAL file /data/data/org.breezyweather/databases/breezyweather.db-wal
04-16 07:01:36.229  7394  7431 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:42.943  7394  7472 E AndroidRuntime: FATAL EXCEPTION: RxCachedThreadScheduler-1
04-16 07:01:42.943  7394  7472 E AndroidRuntime: Process: org.breezyweather, PID: 7394
04-16 07:01:42.943  7394  7472 E AndroidRuntime: P2.e: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at S2.b.a(SourceFile:10)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.observers.c.onError(Unknown Source:13)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onError(Unknown Source:37)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onComplete(Unknown Source:36)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.checkTerminated(SourceFile:61)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.drainNormal(Unknown Source:30)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.n.run(Unknown Source:8)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.q.run(Unknown Source:13)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.q.call(Unknown Source:0)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: Caused by: java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:595)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:480)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at java.math.BigDecimal.<init>(BigDecimal.java:909)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at org.breezyweather.common.extensions.e.G(SourceFile:7)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at y4.b.y(SourceFile:26)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at y4.b.A(SourceFile:918)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at org.breezyweather.sources.p.b(SourceFile:331)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at B0.o.run(SourceFile:141)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at h4.a.run(SourceFile:13)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.e.onComplete(Unknown Source:7)
04-16 07:01:42.943  7394  7472 E AndroidRuntime: 	... 10 more
04-16 07:01:46.372  7537  7537 E g.breezyweather: Not starting debugger since process cannot load the jdwp agent.
04-16 07:01:46.555  7537  7566 E SQLiteLog: (283) recovered 55 frames from WAL file /data/data/org.breezyweather/databases/breezyweather.db-wal
04-16 07:01:46.918  7537  7583 E OpenGLRenderer: Unable to match the desired swap behavior.
04-16 07:01:48.201  7537  7583 E OpenGLRenderer: Unable to match the desired swap behavior.
App version: 5.2.0 (basic, 50200
Android version: 13 (SDK 33); build FP4.TP2D.C.0112.20240313
Device brand: Fairphone
Device manufacturer: Fairphone
Device name: FP4 (FP4eea)
Device model: FP4

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I read the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document and followed instructions if relevant.
- [X] I have updated the app to **[version 5.2.0 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by kosmoz** [comment]
Unfortunately not a very useful stack trace 😕

**Comment by kosmoz** [comment]
Actually, the crash only happens when sending to gadgetbridge is enabled!

**Comment by papjul** [comment]
Decoded stack trace:
```
Caused by: java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
at java.math.BigDecimal.(BigDecimal.java:595)
at java.math.BigDecimal.(BigDecimal.java:480)
at java.math.BigDecimal.(BigDecimal.java:909)
at org.breezyweather.common.extensions.WorkManagerExtensionsKt.org.breezyweather.common.extensions.NumberExtensionsKt.roundDecimals(SourceFile:7)
at org.breezyweather.sources.gadgetbridge.GadgetbridgeService.getAirQuality(SourceFile:26)
                                                              getAirQuality
at org.breezyweather.sources.gadgetbridge.GadgetbridgeService.getDailyForecasts(SourceFile:918)
                                                              getWeatherData
                                                              getWeatherData
at org.breezyweather.sources.RefreshHelper.org.breezyweather.sources.gadgetbridge.GadgetbridgeService.getExtras(SourceFile:331)
                                           broadcastDataIfNecessary
```

Try to use a different air quality source for now, I will check

**Comment by papjul** [comment]
Fixed in d3c03838fb1ec018852932a9ca3ddae67ad53569
